pkg/limits/store.go

```diff
@@ -105,7 +110,23 @@ func (s *usageStore) ForTenant(tenant string, fn iterateFunc) {
 	})
 }
 
-func (s *usageStore) Update(tenant string, streams []*proto.StreamMetadata, lastSeenAt time.Time, cond condFunc) ([]*proto.StreamMetadata, []*proto.StreamMetadata) {
+func (s *usageStore) Update(tenant string, metadata *proto.StreamMetadata, seenAt time.Time) error {
+	if !s.withinActiveWindow(seenAt) {
+		return errOutsideActiveWindow
+	}
+	var (
+		partition    = s.getPartitionForHash(metadata.StreamHash)
+		bucketStart  = seenAt.Truncate(s.bucketSize).UnixNano()
+		bucketCutoff = seenAt.Add(-s.rateWindow).UnixNano()
+	)
+	s.withLock(tenant, func(i int) {
+		s.storeStream(i, tenant, partition, metadata.StreamHash,
+			metadata.TotalSize, seenAt, bucketStart, bucketCutoff)
+	})
+	return nil
+}
+
+func (s *usageStore) UpdateCond(tenant string, streams []*proto.StreamMetadata, lastSeenAt time.Time, cond condFunc) ([]*proto.StreamMetadata, []*proto.StreamMetadata) {
```

Comment:
This is a temporary name, and this method will be refactored further in upcoming changes.